internal/api/http.go

```diff
@@ -187,8 +187,17 @@ func inspectableMIMEType(t string) bool {
 	return strings.HasPrefix(t, "text/") || jsonTypeRE.MatchString(t)
 }
 
-func isValidHost(reqHost, hrtHost string) bool {
-	return strings.Contains(reqHost, hrtHost)
+func isRequestHostValid(reqHost, hrtHost string) bool {
+	reqHost = strings.ToLower(reqHost)
+	hrtHost = strings.ToLower(hrtHost)
+
+	return (reqHost == hrtHost) ||
+		// // Check if the hrtHost is a subdomain of reqHost
+		// // e.g strings.HasSuffix("api.github.com", "".github.com") == true
+		strings.HasSuffix(reqHost, "."+hrtHost) ||
+		// // Check if the reqHost is a subdomain of hrtHost
+		// // e.g strings.HasSuffix("github.com", "".api.github.com") == false
+		strings.HasSuffix(hrtHost, "."+reqHost)
```

Comment:
Let's remove this check; I can't envision a scenario where it would be necessary